### PR TITLE
initial info for contributions

### PR DESCRIPTION
**Pull request by mavimo**
# NB: should be merged AFTER #20 

 - Include MetalLB in order to be able to test ingress without any load balancer service (typically they are expensive 💸)
 - Use port forward to expose ingress (in order to be able to test using Let's encrypt)
 - Some minor fix on missing port in internal service
 - Added DOCS

FYI my next steps will be to add [pebble](https://github.com/letsencrypt/pebble) in order to make easy test on local env the TLS part, but at the moment my focus is on make the TLS part work :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+## Requirements
+
+We will explain how to contribute to this project using a linux machine, in order to be able ot easly contribute you need:
+
+ - A machine with a public IP in order to use let's encrypt (you can provision ad-hoc machine on any clud provider you use)
```
Comment (mholt) [inline CONTRIBUTING.md]:
```suggestion
 - A machine with a public IP in order to use let's encrypt (you can provision ad-hoc machine on any cloud provider you use, or port-forward your home network for dev and testing)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ - A domain that redirect to server IP
```
Comment (mholt) [inline CONTRIBUTING.md]:
```suggestion
- A domain with A and/or AAAA records set to your server's IP
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+We will explain how to contribute to this project using a linux machine, in order to be able ot easly contribute you need:
+
+ - A machine with a public IP in order to use let's encrypt (you can provision ad-hoc machine on any cloud provider you use, or port-forward your home network for dev and testing)
+- A domain with A and/or AAAA records set to your server's IP
+ - [kind](https://github.com/kubernetes-sigs/kind) (to create a development cluster)
+ - [skaffold](https://skaffold.dev/) (to improve development experience)
+ - [Docker HUB](https://hub.docker.com) account (to store your docker images)
+
+## Setup a development cluster
```
Comment (dannav) [inline CONTRIBUTING.md]: This is fine but should probably make it explicit that it is optional. Kubernetes comes with docker for Mac so I don’t want someone getting confused and setting up kind if they don’t need to 😛

File: kubernetes/deploy/01_metallb.yaml [hunk 4]
```diff
@@ -0,0 +1,307 @@
+---
```
Comment (dannav) [inline kubernetes/deploy/01_metallb.yaml]: Same with this file. I would probably put it in an opt folder and link to this manifest

File: CONTRIBUTING.md (hunk 3 above)
Comment (mavimo) [reply CONTRIBUTING.md]: @dannav It's fine for you add a note like:

> OPTIONAL: this is needed only if you don't have a kubernetes cluster to use for development purpose (like Docker for Mac)

File: kubernetes/deploy/01_metallb.yaml (hunk 4 above)
Comment (mavimo) [reply kubernetes/deploy/01_metallb.yaml]: This is required to be able to use a service as `LoadBalancer`, I'm not sure that using an external load balancer should be useful for development purpose (and AFAIK Docker desktop do not support `LoadBalancer` service out of the box).
Do not enable it by default will require extra steps to contributors and only few of them will need to hack the default config.. @dannav WDYT?

Comment (dannav) [reply kubernetes/deploy/01_metallb.yaml]: Yep, you're correct if you want to use a lb :-P. But you can do this with ingress if you want to test locally as well. 

Here's an example w/ minikube and nginx: https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/

I guess the point I'm trying to get across is that unfortunately there are a million ways to do these things :-(. If we want to show one way to do it that's fine. But if we do, we should be explicit that it is optional unless there's a really good reason why it should be done a specific way.

File: CONTRIBUTING.md (hunk 3 above)
Comment (mholt) [reply CONTRIBUTING.md]: @mavimo That looks good to me, sure -- please go for it!

**Pull request merged**